### ISSUE TITLE
Error deserializing decimal.MinValue and decimal.MaxValue, and so on...

### ISSUE THREAD

**Issue by qaqz111** [mgholam/fastJSON#42]
1. Error deserializing `decimal.MinValue` and `decimal.MaxValue`, the result of deserialization is wrong.
2. Lost precision when deserializing `+/-7.9228162514264337593543950335m` to `decimal`, only 17 significant digit perserved.
> The `decimal` has 29 significant digit when top digit is less than '7' according to MSDN
> See [the source code from Microsoft](https://referencesource.microsoft.com/#mscorlib/system/decimal.cs,119) as reference
3. Do not handle `double.NegativeInfinity`, `double.PositiveInfinity`, `float.NegativeInfinity`, `float.PositiveInfinity`.
4. Error deserializing `ulong.MaxValue`. For `ulong` that has a value greater than `long.MaxValue`.
5. Do not handle `IntPtr` and `UIntPtr`. Also, for `UIntPtr`, there is a extreme case too, is that the UIntPtr has a value greater than `long.MaxValue`, it must be treated carefully.

Test code:
``` C#
    public class DigitLimit
    {
        public decimal Mmin = decimal.MinValue; //OK to be serialized but wrongly deserialized to 1
        public decimal Mmax = decimal.MaxValue; //OK to be serialized but wrongly deserialized to -1
        public decimal MminDec = -7.9228162514264337593543950335m; //OK to be serialized but lost precision in deserialization
        public decimal MmaxDec = +7.9228162514264337593543950335m; //OK to be serialized but lost precision in deserialization
        //public double Dmin = double.MinValue; //OK to be serialized but can not be deserialized due to bug of .netFX
        //public double Dmax = double.MaxValue; //OK to be serialized but can not be deserialized due to bug of .netFX
        public double DminDec = -double.Epsilon;
        public double DmaxDec = double.Epsilon;
        public double Dni = double.NegativeInfinity; //OK to be serialized (however without double quote) but when deserializing throw Exception("Could not find token at index " + --index); @NextTokenCore() JSON.cs
        public double Dpi = double.PositiveInfinity; //OK to be serialized (however without double quote) but when deserializing throw Exception("Could not find token at index " + --index); @NextTokenCore() JSON.cs
        public double Dnan = double.NaN;
        public float Fmin = float.MinValue;
        public float Fmax = float.MaxValue;
        public float FminDec = -float.Epsilon;
        public float FmaxDec = float.Epsilon;
        public float Fni = float.NegativeInfinity; //OK to be serialized (however without double quote) but when deserializing throw Exception("Could not find token at index " + --index); @NextTokenCore() JSON.cs
        public float Fpi = float.PositiveInfinity; //OK to be serialized (however without double quote) but when deserializing throw Exception("Could not find token at index " + --index); @NextTokenCore() JSON.cs
        public float Fnan = float.NaN;
        public long Lmin = long.MinValue;
        public long Lmax = long.MaxValue;
        public ulong ULmax = ulong.MaxValue; //OK to be serialized but when deserializing throw System.OverflowException @private object ChangeType(object value, Type conversionType) JSON.cs
        public int Imin = int.MinValue;
        public int Imax = int.MaxValue;
        public uint UImax = uint.MaxValue;

        public IntPtr Iptr1 = new IntPtr(0); //Serialized to a Dict, exception on deserialization
        public IntPtr Iptr2 = new IntPtr(0x33445566); //Serialized to a Dict, exception on deserialization
        public UIntPtr UIptr1 = new UIntPtr(0); //Serialized to a Dict, exception on deserialization
        public UIntPtr UIptr2 = new UIntPtr(0x55667788); //Serialized to a Dict, exception on deserialization
    }
```

Test code:
``` C#
    var oD = new DigitLimit();
    var oDj = FastJSON.JSON.ToNiceJSON(oD);
    var oD_ = FastJSON.JSON.ToObject(oDj);
    var oDj_ = FastJSON.JSON.ToNiceJSON(oD_);
    var cmp = oDj == oDj_; //false;
```
The value of oDj:
``` Json
{
   "$types" : {
      "Test.DigitLimit, SK, Version=1.0.6298.39641, Culture=neutral, PublicKeyToken=null" : "1",
      "System.IntPtr, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" : "2",
      "System.UIntPtr, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" : "3"
   },
   "$type" : "1",
   "Mmin" : -79228162514264337593543950335, //will be deserialized to 1
   "Mmax" : 79228162514264337593543950335, //will be deserialized to -1
   "MminDec" : -7.9228162514264337593543950335, //will be deserialized to -0.123456789012346
   "MmaxDec" : 7.9228162514264337593543950335, //will be deserialized to 0.123456789012346
   //"Dmin" : -1.79769313486232E+308, //OK to be serialized but can't be deserialized due to bug of .netFX
   //"Dmax" : 1.79769313486232E+308, //OK to be serialized but can't be deserialized due to bug of .netFX
   "DminDec" : -4.94065645841247E-324,
   "DmaxDec" : 4.94065645841247E-324,
   //"Dni" : -Infinity, //Exception("Could not find token at index " + --index); @NextTokenCore()
   //"Dpi" : Infinity, //Exception("Could not find token at index " + --index); @NextTokenCore()
   "Dnan" : "NaN",
   "Fmin" : -3.402823E+38,
   "Fmax" : 3.402823E+38,
   "FminDec" : -1.401298E-45,
   "FmaxDec" : 1.401298E-45,
   //"Fni" : -Infinity, //Exception("Could not find token at index " + --index); @NextTokenCore()
   //"Fpi" : Infinity, //Exception("Could not find token at index " + --index); @NextTokenCore()
   "Fnan" : "NaN",
   "Lmin" : -9223372036854775808,
   "Lmax" : 9223372036854775807,
   //"ULmax" : 18446744073709551615, //System.OverflowException @private object ChangeType(object value, Type conversionType) JSON.cs
   "Imin" : -2147483648,
   "Imax" : 2147483647,
   "UImax" : 4294967295//,

   /* IntPtr/UIntPtr is serialized as a struct without it's actual value,
    * but will throw exception on deserialization @private object ChangeType(object value, Type conversionType) JSON.cs
   "Iptr1" : {
      "$type" : "2",
      "Zero" : {
         "$i" : 2
      }
   },
   "Iptr2" : {
      "$type" : "2",
      "Zero" : {
         "$i" : 2
      }
   },
   "UIptr1" : {
      "$type" : "3",
      "Zero" : {
         "$i" : 4
      }
   },
   "UIptr2" : {
      "$type" : "3",
      "Zero" : {
         "$i" : 4
      }
   }*/
}
```
The value of oDj_:
``` Json
{
   "$types" : {
      "Test.DigitLimit, SK, Version=1.0.6298.39003, Culture=neutral, PublicKeyToken=null" : "1"
   },
   "$type" : "1",
   "Mmin" : 1, //not expected!!!
   "Mmax" : -1, //not expected!!!
   "MminDec" : -7.9228162514264335, //not expected!!!
   "MmaxDec" : 7.9228162514264335, //not expected!!!
   "DminDec" : -4.94065645841247E-324,
   "DmaxDec" : 4.94065645841247E-324,
   "Dnan" : "NaN",
   "Fmin" : -3.402823E+38,
   "Fmax" : 3.402823E+38,
   "FminDec" : -1.401298E-45,
   "FmaxDec" : 1.401298E-45,
   "Fnan" : "NaN",
   "Lmin" : -9223372036854775808,
   "Lmax" : 9223372036854775807,
   "Imin" : -2147483648,
   "Imax" : 2147483647,
   "UImax" : 4294967295
}
```

**Comment by mgholam** [comment]
Thanks, I have not handled min max values for most of the types, I will look into it.

**Comment by qaqz111** [comment]
I'm looking forward to new release of RaptorDB which would catch up to the latest fastJSON and fastBinaryJSON. It's the best alternate of the "in-process database sqlite"! Thanks for your great work on RaptorDB!

**Comment by mgholam** [comment]
Most of it is fixed in v2.1.26 except for the following
```c#
// precision loss
public float Fmin; 
public float Fmax;
public decimal MminDec;
public decimal MmaxDec;
```
